### PR TITLE
ANSIENG-3416: Amazon Linux 2023

### PR DESCRIPTION
**Pull request by dhairyam04**
# Description

Jira Link: https://confluentinc.atlassian.net/browse/ANSIENG-3836
Changed molecule scenarios to integrate Amazon Linux 2023.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

[rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel rbac-scram-custom-rhel-fips](https://semaphore.ci.confluent.io/workflows/d1fd2341-8c2a-4cff-b444-3bf85db63eae?pipeline_id=7d92566c-8c59-4882-b589-19e6d6185253)

# Checklist:

- [ ] Any variable/code changes have been validated to be backwards compatible (doesn't break upgrade)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If required, I have ensured the changes can be discovered by cp-ansible discovery codebase
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: galaxy.yml [hunk 1]
```diff
@@ -15,4 +15,4 @@ dependencies: {}
 repository: https://github.com/confluentinc/cp-ansible
 documentation: https://docs.confluent.io/ansible/current/overview.html
 homepage: https://github.com/confluentinc/cp-ansible
-issues: https://github.com/confluentinc/cp-ansible/issues
+issues: https://github.com/confluentinc/cp-ansible/issues
```
Comment (rrbadiani) [inline galaxy.yml:18]: last empty line should not be removed. It causes failures in sanity checks

File: molecule/Dockerfile-amazonlinux2023.j2 [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Install requirements.
+RUN yum -y install rpm \
+ && yum -y update \
+ && yum -y install sudo vim-enhanced \
+ && yum -y install tar procps-ng\
+ && yum install -y systemd \
+ && yum clean all
+
+# Enable systemd
+ENV container docker
```
Comment (rrbadiani) [inline molecule/Dockerfile-amazonlinux2023.j2:14]: line 3 is same

File: molecule/archive-community-plaintext-rhel/molecule.yml [hunk 3]
```diff
@@ -115,4 +115,4 @@ provisioner:
         confluent_cli_archive_url: "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/latest/confluent_latest_linux_${ARCHITECTURE:-amd64}.tar.gz"
         confluent_cli_archive_file_source: "{{ lookup('env', 'MOLECULE_SCENARIO_DIRECTORY') }}/confluent_latest_linux_${ARCHITECTURE:-amd64}.tar.gz"
         confluent_cli_archive_file_remote: false
-        confluent_cli_download_enabled: true
+        confluent_cli_download_enabled: true
```
Comment (rrbadiani) [inline molecule/archive-community-plaintext-rhel/molecule.yml:118]: last empty line should not be removed. It causes failures in sanity checks

**Review by mansisinha — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**